### DISCUSSION TITLE
Github secrets in renovate.json

### DISCUSSION THREAD

**Discussion by duplabe** [renovatebot/renovate#21324] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I would like to use Renovate with my private Terraform modules hosted on Terraform Cloud.

I saw the following errors in Renovate logs:
```
Datasource unauthorized
{
  "baseBranch": "...",
  "datasource": "terraform-module",
  "packageName": "app.terraform.io/...",
  "url": "https://app.terraform.io/..."
}
```

I think I would need to create a host rule in my `renovate.json` in my repository, eg:
```
"hostRules": [
  {
    "hostType": "terraform-module",
    "matchHost": "app.terraform.io",
    "token": TOKEN
  }
]
```

My question is, how can I pass my Terraform Cloud token? Can I use GitHub Secrets? If yes, how could I reference the secret in the json?

### What you tried so far.

Tried to find something related in the documentation without any luck.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
Check out https://docs.renovatebot.com/getting-started/private-packages/ and specifically the `encrypted` part

**Comment by duplabe** [comment]
Thanks rarkins!
